src/core/PhaseVocoder.cpp

```diff
@@ -0,0 +1,273 @@
+/*
+ * PhaseVocoder.cpp - Implementation of the PhaseVocoder class
+ *
+ * Copyright (c) 2023 Daniel Kauss Serna <daniel.kauss.serna@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+#include "PhaseVocoder.h"
+
+#include <algorithm>
+#include <cstdio>
+#include <fftw3.h>
+#include <math.h>
+#include <typeinfo>
+
+#include "lmms_basics.h"
+#include "lmms_constants.h"
+
+namespace lmms {
+
+PhaseVocoder::PhaseVocoder()
+	: m_FFTInput(s_windowSize, 0)
+	, m_IFFTReconstruction(s_windowSize, 0)
+	, m_allMagnitudes(s_windowSize, 0)
+	, m_allFrequencies(s_windowSize, 0)
+	, m_processedFreq(s_windowSize, 0)
+	, m_processedMagn(s_windowSize, 0)
+{
+	m_fftPlan = fftwf_plan_dft_r2c_1d(s_windowSize, m_FFTInput.data(), m_FFTSpectrum.data(), FFTW_MEASURE);
+	m_ifftPlan = fftwf_plan_dft_c2r_1d(s_windowSize, m_FFTSpectrum.data(), m_IFFTReconstruction.data(), FFTW_MEASURE);
+}
+
+PhaseVocoder::~PhaseVocoder()
+{
+	fftwf_destroy_plan(m_fftPlan);
+	fftwf_destroy_plan(m_ifftPlan);
+}
+
+void PhaseVocoder::loadData(const std::vector<float>& originalData, int sampleRate, float newRatio)
```

Comment:
```suggestion
void PhaseVocoder::loadData(std::vector<float> originalData, int sampleRate, float newRatio)
```
`originalData` should be passed by value now that you move at the call sites and here in this method.